### PR TITLE
Add an optional commitHash filter

### PR DESCRIPTION
**Pull request by james-perfectserve**
I like your code, much cleaner than some of the other options I looked through.

For my use case I need the additional option of filtering the artifacts to ones that are tied to a commit hash before matching on name.

fyi, before I touched anything I wasn't able to run `npm ci` something to do with an eslint peer dependency.
I tried installing using pnpm instead of npm and that let me get past the error. Lint isn't part of my normal tool chain, so I didn't spend any time trying to fix it.

### CODE REVIEW

**Comment by alextompkins** [comment]
Thanks for the contribution @james-perfectserve. I'm not sure what the issue was with `npm ci` - I tried it on main and had no issues, but I've bumped all the dependencies on main anyway.

File: src/action.ts [hunk 1]
```diff
@@ -4,6 +4,7 @@ export interface ActionConfig {
   name: string
   path: string
   repo: string
+  commitHash: string
```
Comment (alextompkins) [inline src/action.ts:7]: I realise github will pass in an empty string if this is not provided, but we should make this clearer by marking it as optional.

File: src/action.ts [hunk 2]
```diff
@@ -28,6 +29,7 @@ export function getConfig(): ActionConfig {
     name: core.getInput('name', { required: true }),
     path: core.getInput('path', { required: true }),
     repo: core.getInput('repo', { required: true }),
+    commitHash: core.getInput('commitHash', { required: false }),
```
Comment (alextompkins) [inline src/action.ts:32]: Once we mark `commitHash` as optional, we can set it up by doing 
```ts
commitHash: core.getInput('commitHash', { required: false }) || undefined
```

File: src/main.ts [hunk 3]
```diff
@@ -19,7 +19,12 @@ async function run(): Promise<void> {
       throw Error(`The github API returned a non-success code: ${res.status}`)
     }
 
-    const artifacts = res.data.artifacts
+    let artifacts = res.data.artifacts
+    if (commitHash) {
+      artifacts = artifacts.filter((artifact) =>
+        artifact.workflow_run?.head_sha?.startsWith(commitHash)
```
Comment (alextompkins) [inline src/main.ts:25]: Looks like `octokit`'s type definitions for this response are incomplete. Can we patch the definition with a `.d.ts` file?

**Review by alextompkins — CHANGES_REQUESTED** [review]

File: pnpm-lock.yaml [hunk 4]
```diff
@@ -0,0 +1,4268 @@
+lockfileVersion: 5.4
```
Comment (alextompkins) [inline pnpm-lock.yaml:1]: Please remove the pnpm-lock.yaml, we only want one lockfile in this repo.

**Comment by james-perfectserve** [comment]
> Thanks for the contribution @james-perfectserve. I'm not sure what the issue was with `npm ci` - I tried it on main and had no issues, but I've bumped all the dependencies anyway.

That fixed it, I can run npm ci now

File: src/main.ts (hunk 3 above)
Comment (james-perfectserve) [reply src/main.ts]: I created a d.ts file to temporarily address the issue. Also raised an issue for them https://github.com/actions/toolkit/issues/1137

File: src/action.ts (hunk 2 above)
Comment (james-perfectserve) [reply src/action.ts]: done

File: src/action.ts (hunk 1 above)
Comment (james-perfectserve) [reply src/action.ts]: I updated it to `commitHash?: string`

File: src/main.ts (hunk 3 above)
Comment (alextompkins) [reply src/main.ts]: Thanks for raising the issue :)

**Review by alextompkins — APPROVED** [review]

**Comment by alextompkins** [comment]
I've merged and just released [v1.1.0](https://github.com/alextompkins/download-prior-artifact/releases/tag/v1.1.0). Let me know how that goes for you @james-perfectserve

**Comment by alextompkins** [comment]
I'll update the README too when I get a chance...

**Pull request merged**